floodlight/io/catapultApi.py

```diff
@@ -0,0 +1,495 @@
+            pID_dict.update({group_id: {}})
+        for identifier in sensor_links:
+            if identifier not in pID_dict[group_id]:
+                pID_dict[group_id].update({identifier: []})
+            if athlete_entry[reversed_sensor_mapping[identifier]] not in pID_dict[group_id][identifier]:
+                pID_dict[group_id][identifier].append(athlete_entry[reversed_sensor_mapping[identifier]])
+    
+    framerate = max(max_count, max_count_athlete)
+    pID_dict = dict(sorted(pID_dict.items()))
+
+    timestamps_cs = list(set(t))
+    timestamps_cs.sort()
+
+    print(framerate)
+
+
+    number_of_frames = int((timestamps_cs[-1] - timestamps_cs[0]) / (100 / framerate)) # timestamps are in centiseconds. Magic number 100 is needed for conversion to
+    t_null = timestamps_cs[0]
+
+    return pID_dict, number_of_frames, framerate, t_null
+
+def _get_available_sensor_identifier(pID_dict: Dict[str, Dict[str, List[str]]]) -> str:
+    """
+    Determines an available sensor identifier from the metadata.
+
+    Parameters
+    ----------
+    pID_dict : Dict[str, Dict[str, List[str]]]
+        Metadata dictionary containing player IDs by group.
+
+    Returns
+    -------
+    str
+        The available sensor identifier.
+    """
+    player_identifiers = ["name", "mapped_id", "sensor_id", "number"]
+    available_identifier = [idt for idt in player_identifiers if idt in list(pID_dict.values())[0]]
+    return available_identifier[0]
+
+def create_links_from_meta_data(
+    pID_dict: Dict[str, Dict[str, List[str]]], identifier: str = None
+) -> Dict[str, Dict[str, int]]:
+    """
+    Creates a mapping of player IDs to numerical indices based on the metadata.
+
+    Parameters
+    ----------
+    pID_dict : Dict[str, Dict[str, List[str]]]
+        Metadata dictionary containing player IDs by group.
+    identifier : str, optional
+        The sensor identifier to use for mapping. Defaults to the available identifier.
+
+    Returns
+    -------
+    Dict[str, Dict[str, int]]
+        A dictionary mapping group IDs to player ID indices.
+    """
+    if identifier is None:
+        identifier = _get_available_sensor_identifier(pID_dict)
+
+    links = {}
+    for group_id in pID_dict:
+        links[group_id] = {ID: xID for xID, ID in enumerate(pID_dict[group_id][identifier])}
+
+    return links
+
+def read_position_data_from_dict_list(response_data_dict_list: List[Dict], coord_format: str = "xy") -> List[XY]:
+    """
+    Reads position data from responses data and returns a list of XY objects.
+
+    Parameters
+    ----------
+    response_data_dict_list : List[Dict]
+        List of response_data.
+    coord_format : str, optional
+        The coordinate format ["xy", "latlong"]. Defaults to "xy".
+
+    Returns
+    -------
+    List[XY]
+        A list of XY objects containing position data.
+    
+    Raises
+    ------
+    ValueError
+        If an invalid coordinate format is specified.
+    """
+    mapping = {

```

Comment:
Do we even need the mapping here again? Can you think about a solution where we only need to define the mappings in one function? Then we don't need to check multiple instances if anything changes about those names.

Reply:
changed into using a get_mappings functions which returns the mappings and therefore we only define the mapping once